### PR TITLE
Add personalized movie recommendations to ampcamp4

### PR DESCRIPTION
**Pull request by mengxr**
Add movie recommendation tutorial and code. The code is actually a complete version, tested on a local standalone server. It contains hardcoded locations that I will update later. I'll also do a code style check.

### CODE REVIEW

File: als/scala/MovieLensALS.scala [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    val sc = new SparkContext(
+      "spark://xiangruis-mbp.att.net:7077",
+      "MovieLensALS",
+      "/Users/meng/src/spark-mengxr/dist", 
+      Seq("target/scala-2.10/movielens-als_2.10-0.0.jar")
+    )
+
+    // load ratings and movie titles
+
+    val ratings = sc.textFile("/Users/meng/share/data/movielens/ml-1m/ratings.dat")
```
Comment (etrain) [inline als/scala/MovieLensALS.scala]: Need to take as command line arg. (Same for movies.dat)

Comment (mengxr) [inline als/scala/MovieLensALS.scala]: @etrain, don't spend time on this request now. I will send an update later.

Comment (etrain) [inline als/scala/MovieLensALS.scala]: Ok!

**Comment by mengxr** [comment]
Included a matrix factorization image from @jegonzal .

The code is still a complete version with some hardcoded locations for testing on my local standalone server. Will update the code once I got the scripts from @pwendell .

**Comment by jegonzal** [comment]
Should we merge this now?

**Comment by mengxr** [comment]
It is ready at my side.

**Pull request merged**